src/app/event/[id]/page.tsx

```diff
@@ -17,7 +18,14 @@ type EventPageProps = {
 
 const EventPage: NextPage<EventPageProps> = async ({ params }) => {
   const events = (await getAllEvents()) || [];
-  const event = (await getEvent(params?.id)) || defaultEvent;
+  let event = defaultEvent;
+
+  try {
+    event = (await getEvent(params?.id)) || defaultEvent;
+  } catch (error) {
+    notFound();
+  }
```

Comment:
Would this work with .catch ?

Reply:
Why would I do that? It alreadz async `const EventPage: NextPage<EventPageProps> = async ...,` it makes sense to use a try-catch block instead of .catch, right?

Reply:
If I got it correctly and you can see this ->https://stackblitz.com/edit/vitejs-vite-gzp3un?file=src%2FApp.tsx&terminal=dev

`.catch` is for async errors and promise rejecting and you are correct it kinda belong to the .then es5 code which got replaced in es6. try-catch on the other hand handles SYNC errors as well.

Since the above code is async I think we can change it to

```suggestion
    const event = (await getEvent(params?.id).catch(() => { notFound(); }) || defaultEvent;
```

Reply:
Yas!